templates/ingress-gateways-role.yaml

```diff
@@ -0,0 +1,53 @@
+{{- if .Values.ingressGateways.enabled }}
+
+{{- $root := . }}
+{{- $defaults := .Values.ingressGateways.defaults }}
+
+{{- range .Values.ingressGateways.gateways }}
+{{- $wanAddress := .wanAddress }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ .name }}
+  namespace: {{ $root.Release.Namespace }}
+  labels:
+    app: {{ template "consul.name" $root }}
+    chart: {{ template "consul.chart" $root }}
+    heritage: {{ $root.Release.Service }}
+    release: {{ $root.Release.Name }}
+    component: ingress-gateway
+{{- $source := (default $defaults.wanAddress.source $wanAddress.source) }}
+{{- if (or $root.Values.global.acls.manageSystemACLs $root.Values.global.bootstrapACLs $root.Values.global.enablePodSecurityPolicies (eq $source "Service")) }}
+rules:
+{{- if $root.Values.global.enablePodSecurityPolicies }}
+  - apiGroups: ["policy"]
+    resources: ["podsecuritypolicies"]
+    resourceNames:
+      - {{ .name }}
+    verbs:
+      - use
+{{- end }}
+{{- if (or $root.Values.global.acls.manageSystemACLs $root.Values.global.bootstrapACLs) }}
+  - apiGroups: [""]
+    resources:
+      - secrets
+    resourceNames:
+      - {{ .name }}-ingress-gateway-acl-token
```

Comment:
```suggestion
      - {{ template "consul.fullname" $root }}-{{ .name }}-ingress-gateway-acl-token
```


otherwise I get
```
Error getting Kubernetes secret: secrets "consul-consul-ingress-gateway-ingress-gateway-acl-token" is forbidden: User "system:serviceaccount:default:ingress-gateway" cannot get resource "secrets" in API group "" in the namespace "default"